Lib/dataclasses.py

```diff
@@ -617,9 +617,10 @@ def _frozen_get_del_attr(cls, fields, globals):
     locals = {'cls': cls,
               'FrozenInstanceError': FrozenInstanceError}
     if fields:
-        fields_str = '(' + ','.join(repr(f.name) for f in fields) + ',)'
+        fields_str = '{' + ','.join(repr(f.name) for f in fields) + '}'
     else:
-        # Special case for the zero-length tuple.
+        # Special case for the zero-length set.
+        # Use the empty tuple singleton to avoid unnecessary `set` construction
```

Comment:
Not that it matters much, but the zero length case could just avoid the `or name in ...` test entirely. Maybe fields_str should become fields_test, and then set it to `or name in {<<generated set literal>>}` or set it to an empty string if there are no fields. Then change the generated code to `f'if type(self) is cls {fields_test}:'` Although that doesn't read very well. Maybe tweak fields_test to be something else.

This could be part of a different PR, or include it here. But in any event I'm not positive that the zero length case actually has a test. We should make sure it does for this PR.

Reply:
> But in any event I'm not positive that the zero length case actually has a test. We should make sure it does for this PR.

Added a small test for empty frozen dataclass.